### PR TITLE
LPS-166350

### PR DESCRIPTION
**Pull request by alinedoleron**
https://issues.liferay.com/browse/LPS-166350

**Note:**
This PR is related to the following points:

- Long Labels, Helptext and options break lines
- Input fields texts (text, selects and textarea) truncate with the width of the Input area (25%)
- For fields with truncated text, show tooltips on hover to complement truncated info

### CODE REVIEW

**Comment by alinedoleron** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js [hunk 1]
```diff
@@ -30,6 +31,19 @@ const WORD_CHARACTER_REGEX = /\w/g;
...
+	if (value !== '') {
+		tooltipText = value;
+	}
+	else if (placeholder !== '') {
+		tooltipText = placeholder;
+	}
+
+	return tooltipText !== '' ? {title: tooltipText} : '';
+};
+
```
Comment (cgmonte) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:46]: You could just...

```
const getTooltipTitle = ({placeholder, value}) => {
	let tooltipText = '';

	if (value !== '') {
		tooltipText = {title: value};
	}
	else if (placeholder !== '') {
		tooltipText = {title: placeholder};
	}

	return tooltipText;
};
```
and avoid the ternary :)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Numeric/Numeric.tsx [hunk 2]
```diff
@@ -36,6 +37,25 @@ import type {FieldChangeEventHandler, Locale, LocalizedValue} from '../types';
...
+	if (value !== '') {
+		tooltipText = value;
+	}
+	else if (placeholder !== '') {
+		tooltipText = placeholder;
+	}
+
+	return tooltipText !== '' ? {title: tooltipText} : '';
+};
+
```
Comment (cgmonte) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Numeric/Numeric.tsx:58]: Same as above.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Text/Text.es.js [hunk 3]
```diff
@@ -25,6 +25,19 @@ import {FieldBase} from '../FieldBase/ReactFieldBase.es';
...
+	if (value !== '') {
+		tooltipText = value;
+	}
+	else if (placeholder !== '') {
+		tooltipText = placeholder;
+	}
+
+	return tooltipText !== '' ? {title: tooltipText} : '';
+};
+
```
Comment (cgmonte) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Text/Text.es.js:40]: Same as above. I am starting to think this should extracted to a `utils.js` file. What do you think?

**Review by cgmonte — CHANGES_REQUESTED** [review]
Hi! I made some non-critical suggestions. Please, let me know what you think about them.

**Comment by alinedoleron** [comment]
ci:test:sf

**Review by cgmonte — APPROVED** [review]
LGTM!

**Comment by alinedoleron** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by alinedoleron** [comment]
ci:forward

**Comment by alinedoleron** [comment]
Manually forward to https://github.com/brianchandotcom/liferay-portal/pull/127871

**Pull request closed without merging**